crates/uv-python/src/sysconfig/parser.rs

```diff
@@ -166,18 +166,20 @@ fn parse_string(cursor: &mut Cursor, quote: char) -> Result<String, Error> {
         };
         match c {
             '\\' => {
-                // Handle escaped quotes.
-                if cursor.first() == quote {
-                    // Consume the backslash.
-                    cursor.bump();
-                    result.push(quote);
-                    continue;
-                }
-
-                // Keep the backslash and following character.
-                result.push('\\');
-                result.push(cursor.first());
-                cursor.bump();
+                // Treat the next character as a literal.
+                //
+                // See: https://github.com/astral-sh/ruff/blob/d47fba1e4aeeb18085900dfbbcd187e90d536913/crates/ruff_python_parser/src/string.rs#L194
+                let Some(c) = cursor.bump() else {
+                    return Err(Error::UnexpectedEof);
+                };
+                result.push(match c {
+                    '\\' => '\\',
+                    '\'' => '\'',
+                    '\"' => '"',
+                    _ => {
+                        return Err(Error::UnrecognizedEscape(c));
```

Comment:
I don't think it's necessary for us to handle all the cases here. We generate this file, we know it doesn't contain (e.g.) Unicode names.